cpp/src/strings/regex/config.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "regex.cuh"
+
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/strings/regex/config.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cudf {
+namespace strings {
+namespace detail {
+
+std::pair<std::size_t, size_type> compute_regex_state_memory(strings_column_view const& input,
+                                                             std::string const& pattern,
+                                                             regex_flags const flags,
+                                                             rmm::cuda_stream_view stream)
+{
+  auto d_prog = reprog_device::create(pattern, flags, stream);
```

Comment:
This is a completely ignorant question, but would there ever be any value in exposing the object-oriented interface to the regex machinery? I only ask because the OO exists and therefore must offer us some value internally, but not sure if it would be valuable to callers. 

Similar to our `hash_join` or `groupby` types. The primary value add from those is eliminating redundant work when doing multiple joins/groupbys against the same data.

Reply:
I think not at this time. We may get to a point where we want to separate out the regex-compile step such that our regex APIs do not except a string pattern but rather an object. This way the compiled-object could be reused for multiple columns or even different APIs. This would require a new public class and may not actually be any of the existing ones we use internally.

Reply:
> This way the compiled-object could be reused for multiple columns or even different APIs.

FWIW, in [Java](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html), this is the way.

Reply:
[Python also supports that behavior](https://docs.python.org/3/library/re.html#re.compile). Python users have mostly migrated away from that because Python now also has a built-in LRU cache to avoid recompiling regexes, so using the OO API is only beneficial if you're writing code involving a large number of expressions, but that still constitutes support for that kind of API in libcudf.

Reply:
Also @mythrocks I'm disappointed that "this is the way" didn't link to a meme of some sort...